### PR TITLE
integration test for AWS collections

### PR DESCRIPTION
**Pull request by goneri**

### CODE REVIEW

File: zuul.d/ansible-cloud-jobs.yaml [hunk 1]
```diff
@@ -358,3 +358,30 @@
...
+
+
+### AWS
+- job:
+    name: ansible-test-cloud-integration-aws-py36
+    nodeset: controller-python36
+    dependencies:
+      - name: build-ansible-collection
+    pre-run:
+      - playbooks/ansible-core-ci/pre.yaml
```
Comment (pabelanger) [inline zuul.d/ansible-cloud-jobs.yaml:370]: where is this file?

File: zuul.d/ansible-cloud-jobs.yaml [hunk 2]
```diff
@@ -358,3 +358,30 @@
       - name: github.com/ansible-collections/community.vmware
     vars:
       ansible_collections_repo: github.com/ansible-collections/community.vmware
+
+
+### AWS
+- job:
+    name: ansible-test-cloud-integration-aws-py36
+    parent: ansible-core-ci-aws-session
```
Comment (pabelanger) [inline zuul.d/ansible-cloud-jobs.yaml:366]: so, this is a little tricky. For here, you still need to parent to base-minimal or other 'ansible-test' jobs.

Then add a 2nd, entry
```
- job:
    name: ansible-test-cloud-integration-aws-py36
    parent: ansible-core-ci-aws-session
```
You basically with have 2 'ansible-test-cloud-integration-aws-py36' jobs, with 2 different parents.

Comment (pabelanger) [reply zuul.d/ansible-cloud-jobs.yaml:366]: Another example:

  https://github.com/ansible/network-ee/blob/main/.zuul.d/jobs.yaml#L42
  https://github.com/ansible/network-ee/blob/main/.zuul.d/jobs.yaml#L76

single job, 2 parents.

Comment (goneri) [reply zuul.d/ansible-cloud-jobs.yaml:306]: I actually don't need to inherit from these jobs.

File: zuul.d/project-templates.yaml [hunk 3]
```diff
@@ -53,33 +53,56 @@
...
+        - ansible-test-sanity-docker-stable-2.9
+        - ansible-test-sanity-docker-stable-2.10
+        - ansible-test-units-community-aws-python38
         - build-ansible-collection:
             required-projects:
               - name: github.com/ansible-collections/amazon.aws
               - name: github.com/ansible-collections/ansible.netcommon
               - name: github.com/ansible-collections/ansible.utils
+        - ansible-test-cloud-integration-aws-py36:
+            voting: false
```
Comment (pabelanger) [inline zuul.d/project-templates.yaml:105]: please remove, no need for non-voting jobs in gate. Basically a waste of resources

File: roles/ansible-test-provider/templates/cloud-config-aws.ini.j2 [hunk 4]
```diff
@@ -0,0 +1,11 @@
+[default]
+{% set json_content = lookup('file', ansible_test_provider_passwords_secret_dir + '/aws_session.json')|from_json -%}
+aws_access_key: {{ json_content|json_query('aws.credentials.access_key') }}
+aws_secret_key: {{ json_content|json_query('aws.credentials.secret_key') }}
+security_token: {{ json_content|json_query('aws.credentials.session_token') }}
```
[the remark this answers is not in the document]
Comment (goneri) [reply roles/ansible-test-provider/templates/cloud-config-aws.ini.j2:5]: Yes, this is a temp session. It comes from the ansible-core-ci end-point.

[the remark this answers is not in the document]
Comment (pabelanger) [reply roles/ansible-test-provider/templates/cloud-config-aws.ini.j2:5]: Cool

**Comment by goneri** [comment]
recheck

File: zuul.d/project-templates.yaml (hunk 3 above)
Comment (pabelanger) [reply zuul.d/project-templates.yaml:106]: it looks like this got reverted. We should remove this in gate.

**Pull request merged**